#### Respuesta:

## Problema 1

Se hizo una encuesta a 325 personas mayores de 16 años y se encontró que 120 iban al teatro 
regularmente. Hallar, con un nivel de confianza del 94%, un intervalo para estudiar la proporción de los 
ciudadanos que van al teatro regularmente.

- 1= <b> va al teatro regularmente</b>
- 0=  <b> no va el teatro regularmente</b>

In [58]:
p=120/325
p

0.36923076923076925

Sera la proporcion mediade que los ciudadanos vayan al teatro regularmente

In [59]:
import numpy as np
import scipy.stats
# Creamos la muestra del ejercicio
import random
muestra=[]
for i in range(0,325):
    if i<=179:
        muestra.append(1)
    else:
        muestra.append(0)
        
        

random.shuffle(muestra)



# Lo transformamos
a = 1.0 * np.array(muestra)
a


st.norm.interval(alpha = 0.94 , loc = 0.37, scale = st.sem (a))

C:\Users\Sebastian\AppData\Local\Temp\ipykernel_4700\3803315904.py:23: DeprecationWarning: Use of keyword argument 'alpha' for method 'interval' is deprecated and wil be removed in SciPy 1.11.0. Use first positional argument or keyword argument 'confidence' instead.
  st.norm.interval(alpha = 0.94 , loc = 0.37, scale = st.sem (a))


(0.31805957154820697, 0.421940428451793)

Con un nivel de confianza del 94% se tiene que el intervalo para estudiar la proporción de ciudadanos que van al teatro regularmente esta entre 0.318 y 0.421

## Problema 2

En una encuesta realizada a 800 personas elegidas al azar del censo electoral, 240 declaran su intención de votar al partido A.
- a)    Estima, con un nivel de confianza del  95,45 % ,  entre qué valores se encuentra la intención de voto al susodicho partido en todo el censo.
- b)    Discute, razonadamente, el efecto que tendría sobre el intervalo de confianza el aumento, o la disminución, del nivel de confianza.


- 1= <b> Votar por el partido A </b>
- 0=  <b> No vota por el partido A</b>

In [64]:
# Creamos la muestra del ejercicio
muestra1=[]
for i in range(0,800):
    if i<=239:
        muestra1.append(1)
    else:
        muestra1.append(0)
        
random.shuffle(muestra1)
# Lo transformamos
b= 1.0 * np.array(muestra1)
b


array([0., 1., 0., 0., 1., 0., 0., 1., 0., 0., 1., 0., 0., 1., 0., 0., 1.,
       1., 0., 1., 0., 0., 0., 0., 0., 0., 0., 1., 0., 1., 1., 0., 0., 0.,
       1., 1., 0., 0., 1., 0., 0., 0., 0., 1., 0., 1., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 0., 0., 1., 1., 0., 1., 0., 1., 0., 0., 1., 1.,
       0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 1., 0., 0., 1., 1.,
       0., 0., 1., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 1., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 1., 0.,
       0., 1., 1., 0., 0., 1., 0., 0., 0., 1., 1., 0., 0., 1., 0., 0., 0.,
       0., 0., 0., 1., 0., 1., 1., 1., 1., 0., 0., 0., 0., 0., 0., 1., 0.,
       1., 1., 0., 0., 0., 0., 0., 0., 0., 1., 1., 0., 0., 0., 0., 0., 0.,
       0., 0., 0., 0., 0., 1., 1., 0., 1., 0., 1., 0., 1., 0., 0., 1., 1.,
       0., 1., 1., 0., 0., 0., 1., 0., 0., 0., 1., 1., 0., 1., 0., 0., 1.,
       0., 0., 0., 0., 1., 1., 1., 0., 0., 0., 0., 0., 1., 1., 0., 0., 0.,
       1., 1., 0., 0., 1.

- #### a) Estima, con un nivel de confianza del  95,45 % ,  entre qué valores se encuentra la intención de voto al susodicho partido en todo el censo.

In [65]:
p=240/800
p

0.3

In [77]:
st.norm.interval(confidence = 0.9545 , loc =p, scale = st.sem (muestra1))

(0.2675759855664642, 0.3324240144335357)

Con un nivel de confianza del 95,45% se tiene que el intervalo de la intencion de voto al partido A esta entre 26,75% y 33,24%

- #### b)    Discute, razonadamente, el efecto que tendría sobre el intervalo de confianza el aumento, o la disminución, del nivel de confianza.

#### Aumento

In [78]:
st.norm.interval(confidence = 0.98 , loc =p, scale = st.sem (muestra1))

(0.2622852775628349, 0.3377147224371651)

#### Disminución

In [79]:
st.norm.interval(confidence = 0.90 , loc =p, scale = st.sem (muestra1))

(0.2733336537142725, 0.3266663462857275)

Al aumentar el nivel de confianza el intervalo tambien lo hace. Mientras que si se disminuye la confianza tambien lo hara el intervalo.
Es decir, al tener mas seguridad y por tanto mayor confianza, mayor sera el error.

## Problema 3

En un control de calidad se analizó una muestra aleatoria de  750  tornillos resultando defectuosos  80  de ellos. Hallar: 
- a)    Un intervalo de confianza para la proporción de tornillos defectuosos en el conjunto de producción con  99 %  de confianza. 
- b)    ¿Cuál es el error máximo cometido en la estimación anterior? 
- c)    Si deseamos que el error cometido, con el mismo nivel de confianza, sea la décima parte del apartado anterior, ¿cuál ha de ser el tamaño de la muestra?


- 1= <b> Tornillo defectuoso </b>
- 0=  <b> Tornillo sin defectos</b>

In [75]:
#Proporcion de tornillos defectuosos:
t=80/750
round(t,2)

0.11

In [81]:
# Creamos la muestra del ejercicio
muestra2=[]
for i in range(0,750):
    if i<=79:
        muestra2.append(1)
    else:
        muestra2.append(0)
        
random.shuffle(muestra2)
# Lo transformamos
c= 1.0 * np.array(muestra2)



#### - a)    Un intervalo de confianza para la proporción de tornillos defectuosos en el conjunto de producción con  99 %  de confianza. 

In [82]:
f=st.norm.interval(confidence = 0.99 , loc =t, scale = st.sem (muestra2))
f

(0.07761324400260468, 0.13572008933072865)

Con un nivel de confianza del 99% se tiene que el intervalo de la proporcion de tornillos defectuosos en el conjunto de producción es de 7.76% y 13,57%

#### - b)    ¿Cuál es el error máximo cometido en la estimación anterior?  

In [85]:
error=f[1]-t
error

0.02905342266406198

####  -c)Si deseamos que el error cometido, con el mismo nivel de confianza, sea la décima parte del apartado anterior, ¿cuál ha de ser el tamaño de la muestra?

$$\frac{X}{\sqrt{Muestra}}=Error$$

In [91]:
X=error*pow(750,0.5)
X

0.7956607482919292

In [94]:
print('El valor de X es:'+str(X))

El valor de X es:0.7956607482919292


In [86]:
Error=error*0.1
Error

0.0029053422664061983

Ordenando los valores tenemos:

$$\left( {\frac{X}{Error}}\right)^{2}=Muestra$$

In [96]:
MMuestra=pow((X/Error),2)
print(MMuestra)

74999.99999999999


Para que se cumplan las condiciones, debes er mayor o igual a 74999.99999999999. Si la muestra es de menor tamaño, el error sera de 0,29%